From: Stefan Raspl <raspl@linux.ibm.com>
Subject: net/smc: allow sysctl rmem and wmem defaults for servers
Patch-mainline: v4.18
Git-commit: bd58c7e0860f54710907903ed6daff699d1fc5b9
References: FATE#325698, LTC#167867, bsc#1113481

Description:  smc: Latest upstream fixes and extensions up to 8/17/2018

Upstream-Description:

              net/smc: allow sysctl rmem and wmem defaults for servers

              Without setsockopt SO_SNDBUF and SO_RCVBUF settings, the sysctl
              defaults net.ipv4.tcp_wmem and net.ipv4.tcp_rmem should be the base
              for the sizes of the SMC sndbuf and rcvbuf. Any TCP buffer size
              optimizations for servers should be ignored.

              Signed-off-by: Ursula Braun <ubraun@linux.ibm.com>
              Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Stefan Raspl <raspl@linux.ibm.com>
Acked-by: Petr Tesarik <ptesarik@suse.com>
---
 net/smc/af_smc.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/net/smc/af_smc.c
+++ b/net/smc/af_smc.c
@@ -1122,6 +1122,8 @@ static void smc_tcp_listen_work(struct w
 		sock_hold(lsk); /* sock_put in smc_listen_work */
 		INIT_WORK(&new_smc->smc_listen_work, smc_listen_work);
 		smc_copy_sock_settings_to_smc(new_smc);
+		new_smc->sk.sk_sndbuf = lsmc->sk.sk_sndbuf;
+		new_smc->sk.sk_rcvbuf = lsmc->sk.sk_rcvbuf;
 		sock_hold(&new_smc->sk); /* sock_put in passive closing */
 		if (!schedule_work(&new_smc->smc_listen_work))
 			sock_put(&new_smc->sk);
